File: closure_point.py
def closure_point():
    px = 0
    py = 0

    def setfn(x, y):
        nonlocal px, py
        px = x
        py = y

    def getfn():
        nonlocal px, py
        return (px, py)

    def move():
        nonlocal px, py
        px += 1
        py += 1

    return {'set': setfn, 'get': getfn, 'move': move}


def main():
    my_obj = closure_point()
    my_obj['set'](1, 2)
    print(f"my_obj set at: ", my_obj['get']())
    my_obj['move']()
    print(f"my_obj after move: ", my_obj['get']())

    new_obj = closure_point()
    print(f"new_obj created at: ", new_obj['get']())
    new_obj['move']()
    print(f"new_obj after move: ", new_obj['get']())

    print(f"my_obj still at: ", my_obj['get']())


main()

File: closure_point.go
package main

import "fmt"

func closure_point() map[string] interface{} {
	px := 0
	py := 0

	set := func(x, y int) {
		px = x
		py = y
	}

	get := func() (int, int) {
		return px, py
	}

	move := func() {
		px += 1
		py += 1
	}

	methods := make(map[string] interface{})

	methods["set"] = set
	methods["get"] = get
	methods["move"] = move

	return methods
}

func main() {
	my_obj := closure_point()

	my_obj["set"].(func(int, int))(1, 2)
	fmt.Print("my_obj set at: "); 
	fmt.Println(my_obj["get"].(func() (int, int))())

	my_obj["move"].(func())()
	fmt.Print("my_obj after move: ")
	fmt.Println(my_obj["get"].(func() (int, int))())


	new_obj := closure_point()
	fmt.Print("new_obj created at: ")
	fmt.Println(new_obj["get"].(func() (int, int))())

	new_obj["move"].(func())()
	fmt.Print("new_obj after move: ")
	fmt.Println(new_obj["get"].(func() (int, int))())


	fmt.Print("my_obj still at: ")
	fmt.Println(my_obj["get"].(func() (int, int))())
}
